rspace/src/test/scala/coop/rchain/rspace/history/RadixTreeSpec.scala

```diff
@@ -0,0 +1,679 @@
+
+      _ = inMemoStore.numRecords() shouldBe 3
+
+    } yield ()
+  }
+
+  "function makeActions in work with non-empty tree" should "not create artefacts" in withImplAndStore {
+    (impl, inMemoStore) =>
+      val insertFirstNodesActions = createInsertActions(
+        List(
+          ("111122334455", 0xAA.toByte),
+          ("11112233AABB", 0x11.toByte),
+          ("1111AABBCC", 0x16.toByte)
+        )
+      )
+
+      val insertLastNodesActions = createInsertActions(
+        List(
+          ("33", 0x11.toByte),
+          ("FF0011", 0x22.toByte),
+          ("FF012222", 0x33.toByte)
+        )
+      )
+      for {
+        rootNode1 <- impl.loadNode(RadixHistory.emptyRootHash.bytes, noAssert = true)
+
+        //  Create tree with 3 leafs
+        rootNode2Opt <- impl.makeActions(rootNode1, insertFirstNodesActions)
+        committed1   <- impl.commit
+
+        nodesCount1 = inMemoStore.numRecords()
+        printed = println(
+          s"Nodes count after creating tree of 3 leafs (without root node) : ${nodesCount1.toString}"
+        )
+
+        printedTree1 <- impl.printTree(rootNode2Opt.get, "TREE1", true)
+
+        //  Append in existing tree 3 leafs...
+        rootNode3Opt <- impl.makeActions(rootNode2Opt.get, insertLastNodesActions)
+        committed2   <- impl.commit
+        nodesCount2  = inMemoStore.numRecords()
+        printed2 = println(
+          s"Nodes count after appending ${insertLastNodesActions.size.toString} leafs (without root node): ${nodesCount2.toString}"
+        )
+        printedTree2 <- impl.printTree(rootNode3Opt.get, "TREE2", true)
+
+        _ = nodesCount1 shouldBe 2
+        _ = nodesCount2 shouldBe 3
+      } yield ()
+  }
+
+  "function saveNode" should "put node into store" in withImplAndStore { (impl, inMemoStore) =>
+    val itemData = createBV32(0xCB.toByte)
+    val key      = TestData.hexKey("0123456F1").toVector
+    for {
+      itemOpt     <- impl.update(RadixTree.EmptyItem, key, itemData)
+      nodesCount1 = inMemoStore.numRecords()
+
+      node     <- impl.constructNodeFromItem(itemOpt.get)
+      saved    = impl.saveNode(node)
+      commited <- impl.commit
+
+      //  After saving node numRecords must return 1
+      nodesCount2 = inMemoStore.numRecords()
+      _           = nodesCount1 shouldBe 0
+      _           = nodesCount2 shouldBe 1
+    } yield ()
+  }
+
+  "sequentialExport" should "export all data from tree" in withImplAndStore { (impl, store) =>
+    val insertFirstNodesActions = createInsertActions(
+      List(
+        ("111122334455", 0xAA.toByte),
+        ("11112233AABB", 0x11.toByte),
+        ("1111AABBCC", 0x16.toByte),
+        ("33", 0x11.toByte),
+        ("FF0011", 0x22.toByte),
+        ("FF012222", 0x33.toByte)
+      )
+    )
+
+    val exportSettings = ExportDataSettings(
+      flagNodePrefixes = false,
+      flagNodeKeys = true,
+      flagNodeValues = true,
+      flagLeafPrefixes = false,
+      flagLeafValues = true
```

Comment:
Please set all flags, and validate that all exported data is correct. For validating LeafPrefixes and LeafData you can just compare this value with initial data

Reply:
To validate NodePrefixes, NodeKeys and NodeValues don't use hardcoded values. Use validation procedure as here: https://github.com/hilltracer/rchain/blob/709e7fa710df0ef40b4a5723a2125709b533283f/node/src/test/scala/coop/rchain/node/perf/HistoryGenKeySpec.scala#L280-L435

Make the first export, then create storage from the exported data, and repeat the export from the new storage. If the second export is successful - this will be proof of the correctness of the node data (without hardcoded value).